x/supplier/keeper/msg_server_stake_supplier.go

```diff
@@ -105,9 +113,10 @@ func (k msgServer) StakeSupplier(ctx context.Context, msg *types.MsgStakeSupplie
 		supplier.UnstakeSessionEndHeight = sharedtypes.SupplierNotUnstaking
 	}
 
-	// MUST ALWAYS stake or upstake (> 0 delta)
-	if coinsToEscrow.IsZero() {
-		err = types.ErrSupplierInvalidStake.Wrapf("Signer %q must escrow more than 0 additional coins", msg.Signer)
+	// MUST ALWAYS stake or upstake (>= 0 delta)
+	// TODO_CONSIDERATION: Should we allow stake decrease down to min stake?
```

Comment:
I think this would improve the supplier UX. :+1: 

I think we might need slightly more than the existing services activation height map logic to ensure that partial unstaking only takes effect after sessions pending settlement have been settled (and/or limit the unstake amount to `stake - maxSlashing`).